### ISSUE TITLE
configuration options for length field of length prefix framing

### ISSUE THREAD

**Issue by weberdaniel** [actor-framework/actor-framework#1896]
Some length prefixed protocols don't write the payload size into the length field of a frame, but use the length field to denote the total size of the frame (either payload_size+size_of_length_field or just payload_size). For convenience the implementation of a configuration option for this might be considered. While  it is possible to solve this issue by implementing a custom framing class by e.g. using copy/paste, deriving from the framing class and then altering the constraints of the base class in the derived class, might result in a LSP violation according to "C++ Software Design" (Klaus Iglberger).

**Comment by Neverlord** [comment]
Certainly makes sense to have this configurable. An option to configure the size of the length field might be a good idea as well while we're at it. Currently, we always assume a 32-bit field. Not sure if we can still squeeze this in for 1.0, but we'll have it on the radar.

**Comment by weberdaniel** [comment]
The configuration option might be convenient for people who try it out for the first time. Most space communication protocols are standardized by CCSDS (https://public.ccsds.org/). Examples would be the TC Space Data Link Protocol (CCSDS 232.0-B-4) or TM Space Data Link Protocol (CCSDS 132.0-B-3 ). Most have a fixed size header with the length field on an arbitrary position within the header. Some also support additional secondary headers to be activated or not (for example in telemetry). In such specialized cases customizing/adopting the length-prefix framing case is probably a necessity anyway. That being said, sometimes it may be necessary to convert data units from TCP/IP to UDP/IP especially for robotic real-time applications.

**Comment by dominiklohmann** [comment]
I tried using the length prefix framing for size-prefixed FlatBuffers, but ran into the issue that the length is currently assumed to always be 32 bit instead of 64 bit. Are there any plans to make this configurable?

**Comment by Neverlord** [comment]
> Are there any plans to make this configurable?

Yes. However, making the size configurable will break the ABI because we need to change `config_type` (and thus the type of the member variable) in `caf/net/lp/with.hpp`. So the next window for this feature is 2.0.

**Comment by weberdaniel** [comment]
thanks for the help, i actually wanted to share some positive experiences with CAF 0.19.3 on a conference, but workload is too high.
